desktop/components/react/infinite_scroll/InfiniteScroll.js

```diff
@@ -0,0 +1,64 @@
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+import { throttle } from 'underscore'
+
+const THROTTLE_MS = 200
+
+export default class InfiniteScroll extends Component {
+  static propTypes = {
+    children: PropTypes.node.isRequired,
+    onTrigger: PropTypes.func.isRequired,
+    triggerElement: PropTypes.string.isRequired
+  }
+
+  state = {
+    isMounted: false,
+    $node: {}
+  }
+
+  componentDidMount () {
+    const { triggerElement } = this.props
+
+    this.$ = require('jquery')
+    this.$(window).on(`scroll${triggerElement}`, this.handleScroll())
+
+    this.setState(() => {
+      return {
+        isMounted: true,
+        $node: this.$(triggerElement)
+      }
+    })
+  }
+
+  componentWillUnmount () {
+    this.$(window).off(`scroll${this.props.triggerElement}`)
+  }
+
+  handleScroll = () => {
+    const { onTrigger } = this.props
+
+    return throttle(() => {
+      const { isMounted, $node } = this.state
+
+      if (!isMounted) {
+        return
+      }
+
+      // TODO: Add ability to pass in threshold
+      const threshold = this.$(window).height() + this.$(window).scrollTop() + 500 // px push
```

Comment:
500px, which gives the view on desktop and mobile a good amount of padding to trigger the fetch before you scroll completely to the bottom. Am going to sweep back through per the `TODO` above and add the ability to pass in 
a custom threshold.